### PR TITLE
Initial frontend setup

### PR DESCRIPTION
**Pull request by dmijatovic**
# Initial setup frontend with NextJS, MUI-5, Tailwind and Jest

Adresses #42 #43 #44 

How to test:
- in the **frontend folder** copy `env.local.example` to `env.local`. The content of this file is send to you via Teams or email. If not, ask one of core developers to provide you content of env.local file.
- `docker-compose up --build` in the root of the project should start all parts of RSD. On `http://localhost:3000` you should see home page (image below). To stop use `Ctrl +C`
-  you can navigate to: software and project pages (these have only page title)
-  you can try to login with SURFconext `test environment`:  when redirected to SURF login page select `SURFconext Test Idp` provider and specify user `professor1` and pass `professor1`. After authentication you will be send back to user/profile page by default.  
- unit testing with Jest: see readme file in the frontend directory for more info. **You need Node.js and Yarn installed locally on your machine**. When running docker-compose the tests are runned in the container before the build.

## Expected result after docker-compose up
image

### CODE REVIEW

File: frontend/.gitignore [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
Comment (ewan-escience) [inline frontend/.gitignore:1]: Please move its contents to the toplevel .gitignore and remove this file

File: docker-compose.yml [hunk 2]
```diff
@@ -2,26 +2,71 @@ version: "3.0"
...
+    container_name: frontend
+    build:
+      context: ./frontend
+      # dockerfile to use for build
+      dockerfile: Dockerfile
+    # update version number to corespond to frontend/package.json
+    image: rsd/frontend:0.0.4
+    env_file:
+      # configuration
+      - ./frontend/.env.production.local
```
Comment (ewan-escience) [inline docker-compose.yml:55]: This is the local development docker-compose file, please use the file env.local

File: frontend/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+version: '3'
```
Comment (ewan-escience) [inline frontend/docker-compose.yml:1]: Please remove this file, you can start without the frontend with --scale

**Review by jmaassen — APPROVED** [review]
Works as expected!

**Pull request merged**